services/frontend-service/src/ui/Pages/Home/Home.test.tsx

```diff
@@ -32,20 +32,278 @@ describe('App', () => {
+                    team: 'foo',
+                },
+            },
+            expect: (teamNames) => expect(teamNames).toStrictEqual(['<No Team>', 'foo', 'test']),
+        },
+    ];
+
+    describe.each(data)(`Renders an Application Card`, (testcase) => {
+        it(testcase.name, () => {
+            // given
+            UpdateOverview.set({ applications: testcase.applications, environments: {} });
+            // when
+            const teamNames = renderHook(() => useTeamNames()).result.current;
+            testcase.expect(teamNames);
+        });
+    });
+});
+
+describe('Get applications from selected teams (useFilteredApps)', () => {
+    interface dataT {
+        name: string;
+        selectedTeams: string[];
+        applications: { [key: string]: Application };
+        expect: (teamNames: number) => void;
+    }
+
+    const data: dataT[] = [
+        {
+            name: 'gets every app - 2 results',
+            selectedTeams: ['dummy', 'foo'],
+            applications: {
+                foo: {
+                    name: 'foo',
+                    releases: [],
+                    sourceRepoUrl: 'http://foo.com',
+                    team: 'dummy',
+                },
+                bar: {
+                    name: 'bar',
+                    releases: [],
+                    sourceRepoUrl: 'http://bar.com',
+                    team: 'test',
+                },
+                example: {
+                    name: 'example',
+                    releases: [],
+                    sourceRepoUrl: 'http://example.com',
+                    team: 'test2',
+                },
+                team: {
+                    name: 'team',
+                    releases: [],
+                    sourceRepoUrl: 'http://team.com',
+                    team: 'foo',
+                },
+            },
+            expect: (teamNames) => expect(teamNames).toStrictEqual(2),
+        },
+        {
+            name: 'shows both applications of the selected team - 2 results',
+            selectedTeams: ['dummy'],
+            applications: {
+                foo: {
+                    name: 'foo',
+                    releases: [],
+                    sourceRepoUrl: 'http://foo.com',
+                    team: 'dummy',
+                },
+                bar: {
+                    name: 'bar',
+                    releases: [],
+                    sourceRepoUrl: 'http://bar.com',
+                    team: 'dummy',
+                },
+                team: {
+                    name: 'team',
+                    releases: [],
+                    sourceRepoUrl: 'http://team.com',
+                    team: 'foo',
+                },
+            },
+            expect: (teamNames) => expect(teamNames).toStrictEqual(2),
+        },
+        {
+            name: 'no teams selected (shows every application) - 4 results',
+            selectedTeams: [],
+            applications: {
```

Comment:
is this `applications` block the same as above?
If so, extract it as a constant.
Then it's easier to see that the 2 tests use the same data.

Reply:
In this app block the teams `foo` and `example` don't have teams assigned to them. This is meant to test if every application shows up when there are no selected teams (even the apps with no team assigned to them).
In the app blocks above every app has a team.